File: app/core/portal_role_home.py
"""مسیر ورود پنل به ازای User.role — منبع واحد برای /api/auth/home و redirect فرانت."""

from __future__ import annotations

from urllib.parse import urlencode

# نقش کمیته → kind مسیر
_COMMITTEE_ROLE_TO_KIND: dict[str, str] = {
    "progress_committee": "progress",
    "education_committee": "education",
    "deputy_education": "education",
    "supervision_committee": "supervision",
    "monitoring_committee_officer": "supervision",
    "specialized_commission": "supervision",
    "therapy_committee_chair": "therapy",
    "therapy_committee_executor": "therapy",
}

STAFF_DEFAULT_LANE = "admissions"

_STAFF_LANE_BY_ASSIGNED: dict[str, str] = {}
for _lane, _roles in {
    "admissions": ("admissions_officer", "admission_officer", "interviewer"),
    "instruction": ("instructor", "teaching_assistant", "teaching_assistant_or_instructor"),
    "therapy-coord": ("therapy_education_coordinator",),
    "course-committee": (
        "course_committee",
        "course_committee_scientific",
        "course_committee_executive",
        "scientific_officer_course_committee",
    ),
}.items():
    for _r in _roles:
        _STAFF_LANE_BY_ASSIGNED[_r] = _lane

_ASSIGNED_TO_COMMITTEE_KIND: dict[str, str] = {
    "committee": "progress",
    "progress_committee": "progress",
    "progress_committee_project": "progress",
    "education_committee": "education",
    "deputy_education": "education",
    "deputy_education_director": "education",
    "course_committee_executive": "education",
    "scientific_officer_course_committee": "education",
    "supervision_committee": "supervision",
    "monitoring_committee_officer": "supervision",
    "specialized_commission": "supervision",
    "therapy_committee_chair": "therapy",
    "therapy_committee_executor": "therapy",
}


def staff_lane_for_assigned_role(code: str | None) -> str:
    if not code:
        return STAFF_DEFAULT_LANE
    return _STAFF_LANE_BY_ASSIGNED.get(code.strip(), STAFF_DEFAULT_LANE)


def committee_kind_for_assigned_role(code: str | None) -> str:
    if not code:
        return "progress"
    return _ASSIGNED_TO_COMMITTEE_KIND.get(code.strip(), "progress")


def committee_kind_for_role(role: str | None) -> str:
    if not role:
        return "progress"
    return _COMMITTEE_ROLE_TO_KIND.get(role.strip(), "progress")


def staff_lane_path(lane: str = STAFF_DEFAULT_LANE) -> str:
    return f"/panel/portal/staff/{lane}"


def committee_kind_path(kind: str | None = None, role: str | None = None) -> str:
    k = kind or committee_kind_for_role(role)
    return f"/panel/portal/committee/{k}"


# (path, default_tasks_tab) — tab=None یعنی بدون query
PORTAL_ROLE_HOME: dict[str, tuple[str, str | None]] = {
    "student": ("/panel/portal/student", None),
    "therapist": ("/panel/portal/therapist", "pending"),
    "supervisor": ("/panel/portal/supervisor", "reviews"),
    "staff": (staff_lane_path(STAFF_DEFAULT_LANE), "pending"),
    "site_manager": ("/panel/portal/site-manager", "pending"),
    "interviewer": ("/panel/portal/interviewer", None),
    "finance": ("/panel/finance", None),
    "progress_committee": (committee_kind_path("progress"), "reviews"),
    "education_committee": (committee_kind_path("education"), "reviews"),
    "supervision_committee": (committee_kind_path("supervision"), "reviews"),
    "specialized_commission": (committee_kind_path("supervision"), "reviews"),
    "therapy_committee_chair": (committee_kind_path("therapy"), "reviews"),
    "therapy_committee_executor": (committee_kind_path("therapy"), "reviews"),
    "deputy_education": (committee_kind_path("education"), "reviews"),
    "monitoring_committee_officer": (committee_kind_path("supervision"), "reviews"),
    "admin": ("/panel", None),
}


def portal_home_path(role: str | None) -> str | None:
    """مسیر پایه پنل؛ None اگر نقش در نقشه نیست."""
    if not role:
        return None
    entry = PORTAL_ROLE_HOME.get(role.strip())
    return entry[0] if entry else None


def default_tasks_tab_for_role(role: str | None) -> str | None:
    """تب کارتابل پیش‌فرض: pending | reviews | None."""
    if not role:
        return None
    entry = PORTAL_ROLE_HOME.get(role.strip())
    return entry[1] if entry else None


def redirect_url_for_role(role: str | None) -> str:
    """
    URL کامل برای redirect بعد از login یا /panel.
    نقش ناشناخته → /panel (داشبورد عمومی).
    """
    if not role:
        return "/panel"
    entry = PORTAL_ROLE_HOME.get(role.strip())
    if not entry:
        return "/panel"
    path, tab = entry
    if tab:
        return f"{path}?{urlencode({'tab': tab})}"
    return path

File: admin-ui/src/utils/portalRoleHome.js
/**
 * مسیر ورود پنل به ازای User.role — هم‌تراز با app/core/portal_role_home.py
 */
import { committeeKindForPortalRole, getCommitteeKindPath } from './portalCommitteeKinds'
import { getStaffLanePath } from './portalStaffLanes'

const STAFF_DEFAULT_LANE = 'admissions'

/** @type {Record<string, { path: string, tasksTab: string | null }>} */
export const PORTAL_ROLE_HOME = {
  student: { path: '/panel/portal/student', tasksTab: null },
  therapist: { path: '/panel/portal/therapist', tasksTab: 'pending' },
  supervisor: { path: '/panel/portal/supervisor', tasksTab: 'reviews' },
  staff: { path: getStaffLanePath(STAFF_DEFAULT_LANE), tasksTab: 'pending' },
  course_committee: { path: getStaffLanePath('course-committee'), tasksTab: 'pending' },
  teaching_assistant: { path: getStaffLanePath('instruction'), tasksTab: 'pending' },
  site_manager: { path: '/panel/portal/site-manager', tasksTab: 'pending' },
  interviewer: { path: '/panel/portal/interviewer', tasksTab: null },
  finance: { path: '/panel/finance', tasksTab: null },
  progress_committee: { path: getCommitteeKindPath('progress'), tasksTab: 'reviews' },
  education_committee: { path: getCommitteeKindPath('education'), tasksTab: 'reviews' },
  supervision_committee: { path: getCommitteeKindPath('supervision'), tasksTab: 'reviews' },
  specialized_commission: { path: getCommitteeKindPath('supervision'), tasksTab: 'reviews' },
  therapy_committee_chair: { path: getCommitteeKindPath('therapy'), tasksTab: 'reviews' },
  therapy_committee_executor: { path: getCommitteeKindPath('therapy'), tasksTab: 'reviews' },
  deputy_education: { path: getCommitteeKindPath('education'), tasksTab: 'reviews' },
  monitoring_committee_officer: { path: getCommitteeKindPath('supervision'), tasksTab: 'reviews' },
  admin: { path: '/panel', tasksTab: null },
}

const PORTAL_LABELS_FA = {
  student: 'پنل آموزشی',
  therapist: 'پنل درمانگر',
  supervisor: 'پنل سوپروایزر',
  staff: 'پنل پذیرش',
  course_committee: 'پنل کمیته دروس',
  teaching_assistant: 'پنل مدرس و کمک‌مدرس',
  site_manager: 'پنل مسئول سایت',
  interviewer: 'پنل مصاحبه‌گر',
  finance: 'داشبورد مالی',
  progress_committee: 'کمیته پیشرفت',
  education_committee: 'کمیته آموزش',
  supervision_committee: 'کمیته نظارت',
  specialized_commission: 'کمیته نظارت',
  therapy_committee_chair: 'کمیته درمان',
  therapy_committee_executor: 'کمیته درمان',
  deputy_education: 'کمیته آموزش',
  monitoring_committee_officer: 'کمیته نظارت',
}

const PORTAL_ICONS = {
  student: '🎓',
  therapist: '💊',
  supervisor: '👁️',
  staff: '🏢',
  course_committee: '📚',
  teaching_assistant: '🎓',
  site_manager: '🏗️',
  interviewer: '🎤',
}

/** @param {string | null | undefined} role */
export function getPortalHomePath(role) {
  const entry = role ? PORTAL_ROLE_HOME[role] : null
  return entry?.path ?? null
}

/** @param {string | null | undefined} role @returns {'pending' | 'reviews' | null} */
export function getPortalDefaultTasksTab(role) {
  const entry = role ? PORTAL_ROLE_HOME[role] : null
  return entry?.tasksTab ?? null
}

/** @param {string | null | undefined} role */
export function getPortalHomeHref(role) {
  const path = getPortalHomePath(role)
  if (!path) return '/panel'
  const tab = getPortalDefaultTasksTab(role)
  if (tab) {
    return `${path}?tab=${encodeURIComponent(tab)}`
  }
  return path
}

/** @param {string | null | undefined} role */
export function getPortalQuickLink(role) {
  const path = getPortalHomePath(role)
  if (!path || role === 'admin') return null
  return {
    path: getPortalHomeHref(role),
    label: PORTAL_LABELS_FA[role] || 'پنل نقش',
    icon: PORTAL_ICONS[role] || '📋',
  }
}

/** مسیر committee kind برای redirect از /panel/portal/committee */
export function getCommitteeHomePathForRole(role) {
  const kind = committeeKindForPortalRole(role)
  return getCommitteeKindPath(kind)
}

